pkg/scheduler/cache/cache.go

```diff
@@ -270,30 +270,35 @@ func (cache *schedulerCache) getPodResizeRequirements(pod *v1.Pod, resizeRequest
 	return resizeContainersMap, podResource, nil
 }
 
-func restoreResources(pod *v1.Pod, restoreResources string) error {
+func (cache *schedulerCache) restoreResources(oldPod, newPod *v1.Pod, restoreResources string) error {
+	podKey, _ := getPodKey(oldPod)
+	currPodState, _ := cache.podStates[podKey]
+	cachedPod := currPodState.pod
 	restoreContainersMap := make(map[string]v1.Container)
 	if err := json.Unmarshal([]byte(restoreResources), &restoreContainersMap); err != nil {
-		glog.Errorf("Pod %s unmarshalling restore resource annotation '%s' failed. Error: %v", pod.Name, restoreResources, err)
+		glog.Errorf("Pod %s unmarshalling restore resource annotation '%s' failed. Error: %v", newPod.Name, restoreResources, err)
 		return err
 	}
-	for i, container := range pod.Spec.Containers {
+	for i, container := range newPod.Spec.Containers {
 		if restoreContainer, ok := restoreContainersMap[container.Name]; ok {
 			if restoreContainer.Resources.Requests != nil {
 				for k, v := range restoreContainer.Resources.Requests {
-					pod.Spec.Containers[i].Resources.Requests[k] = v
+					newPod.Spec.Containers[i].Resources.Requests[k] = v
```

Comment:
restoreContainer should contain all the resources. not just the ones that changed. Seems like there a deep copy of the Requests would be good enough. no need to iterate. same with the limit below.

Reply:
yes. i'll fix this.